zerver/lib/saved_snippets.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any
+
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import ResourceNotFoundError
+from zerver.models import SavedSnippet, UserProfile
+from zerver.tornado.django_api import send_event_on_commit
+
+
+def do_create_saved_snippet(
+    title: str,
+    content: str,
+    user_profile: UserProfile,
+) -> SavedSnippet:
+    saved_snippet = SavedSnippet.objects.create(
+        realm=user_profile.realm,
+        user_profile=user_profile,
+        title=title,
+        content=content,
+    )
+
+    event = {
+        "type": "saved_snippets",
+        "op": "add",
+        "saved_snippet": saved_snippet.to_api_dict(),
+    }
+    send_event_on_commit(user_profile.realm, event, [user_profile.id])
+
+    return saved_snippet
```

Comment:
This should use `@transaction.atomic(durable=True)`, and probably have a `RealmAuditLog` entry; we've usually regretted adding new data without using that.

Also, this file should probably be `zerver/actions/saved_snippets.py`, given the type of functions you have in here.

Reply:
I have moved the file to `actions` folder and added a `RealmAuditLog` entry for add function.